### PR TITLE
fix: support all whichkey methods

### LINKED ISSUE #81: Cannot parse which-key mappings
I have my which-key mappings table setup like the Primary example in the which-key [README](https://github.com/folke/which-key.nvim#%EF%B8%8F-mappings)

Similar to this...
```lua
wk.register({
  f = {
    name = "file", -- optional group name
    f = { "<cmd>Telescope find_files<cr>", "Find File" }, -- create a binding with label
    r = { "<cmd>Telescope oldfiles<cr>", "Open Recent File", noremap=false, buffer = 123 }, -- additional options for creating the keymap
    n = { "New File" }, -- just a label. don't create any mapping
    e = "Edit File", -- same as above
    ["1"] = "which_key_ignore",  -- special label to hide it in the popup
    b = { function() print("bar") end, "Foobar" } -- you can also pass functions!
  },
}, { prefix = "<leader>" })
```

Using `prefix` at the end... and I get an error when I run `HawtkeysAll` that says  "Error parsing which-key table"

I have this as part of my config opts for hawtkeys

```lua
-- If you use whichkey.register with an alias eg wk.register
        ["wk.register"] = {
            method = "which_key",
        },
```

### PR DESCRIPTION
**Pull request by tris203**
closes #81 

needs further work on method1

I don't really understand the structure though as the <leader> section just gets dropped seemingly

@willothy any ideas?

### CODE REVIEW

**Comment by tris203** [comment]
Minimal repro here

```lua
local wk = require("which-key.mappings")

local str = [[
 return {
     w = {
         name = "file", -- optional group name
         o = { ":lua print('hello')<CR>", "Find File" }, -- create a binding with label
         n = {
             function()
                 print("bar")
             end,
             "Foobar",
         }, -- you can also pass functions!
     },
 }, { prefix = "<leader>" }
]]

-- local str = [[
--  return ({
--      ["<leader>wf"] = { ':lua print("hello")<CR>', "hello" },
--  })
-- ]]

local obj = loadstring(str)()

print(vim.inspect(obj))

local wkd = wk.parse(obj)

print(vim.inspect(wkd))

```

**Comment by tris203** [comment]
Okay, i think the issue is fixed, but I don't know why the test for the which-key message is now failing....

*edit, that took me an embarrassingly long time to fix...

To explain more there was a few issues at play.
First wrapping the return in params caused it not to return the subobjects properly, then additionally the object needs to be wrapped in a table and unpacked, as otherwise return in the loadstring just returns the first table, not both

File: lua/hawtkeys/ts.lua [hunk 1]
```diff
@@ -344,7 +344,9 @@ local function find_maps_in_file(filePath)
                 local strObj =
                     vim.treesitter.get_node_text(node.node, fileContent)
                 local ok, tableObj = pcall(function()
-                    return loadstring("return " .. strObj)()
+                    --Remove wrapping parens issue #81
```
Comment (tris203) [inline lua/hawtkeys/ts.lua]:
```suggestion
                    --Remove wrapping parens and wrap in table and unpack - issue #81
```

**Review by tris203 — COMMENTED** [review]
clearer comment

**Review by willothy — APPROVED** [review]
lgtm - loadstring can definitely be tricky, it would be good if we could figure out an alternative at some point though I'm not sure what that would be.

**Pull request merged**